src/RelationViewer/RelationViewerProvider.ts

```diff
@@ -0,0 +1,135 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as vscode from 'vscode';
+
+import {getRelationData} from './example/RelationExample';
+import {RelationViewer} from './RelationViewer';
+
+export class RelationViewerDocument implements vscode.CustomDocument {
+  private readonly _uri: vscode.Uri;
+  private _metadataViwer: RelationViewer[];
+
+  static async create(uri: vscode.Uri):
+      Promise<RelationViewerDocument|PromiseLike<RelationViewerDocument>> {
+    return new RelationViewerDocument(uri);
+  }
+
+  private constructor(uri: vscode.Uri) {
+    this._uri = uri;
+    this._metadataViwer = [];
+  }
+
+  public get uri() {
+    return this._uri;
+  }
+
+  // CustomDocument implements
+  dispose(): void {
+    // NOTE panel is closed before document and this is just for safety
+    this._metadataViwer.forEach((view) => {
+      while (this._metadataViwer.length) {
+        view.disposeMetadataView();
+      }
+    });
+    this._metadataViwer = [];
+  }
+
+  public openView(panel: vscode.WebviewPanel, extensionUri: vscode.Uri, fileUri: vscode.Uri) {
+    let view = new RelationViewer(panel, extensionUri);
+
+    view.initWebview();
+    view.loadContent();
+    this._metadataViwer.push(view);
+
+    const payload = getRelationData(fileUri);
+
+    // Send a message the relation data to the web view
+    panel.webview.postMessage({type: 'create', payload: payload});
+
+    panel.onDidDispose(() => {
+      // TODO make faster
+      this._metadataViwer.forEach((view, index) => {
+        if (view.owner(panel)) {
+          view.disposeMetadataView();
+          this._metadataViwer.splice(index, 1);
+        }
+      });
+    });
+
+    return view;
+  }
+}
+
+export class RelationViewerProvider implements
+    vscode.CustomReadonlyEditorProvider<RelationViewerDocument> {
+  public static readonly viewType = 'one.viewer.relation';
+  private _context: vscode.ExtensionContext;
+
+  public static register(context: vscode.ExtensionContext): void {
+    const provider = new RelationViewerProvider(context);
+
+    const registrations = [
+      vscode.window.registerCustomEditorProvider(RelationViewerProvider.viewType, provider, {
+        webviewOptions: {
+          retainContextWhenHidden: true,
+        }
+      }),
+      vscode.commands.registerCommand(
+          'one.viewer.relation.show',
+          async (uri) => {
+            // If the method is executed in the ONE Explorer, change the uri instance.
+            let fileUri = uri.uri ? uri.uri : uri;
```

Comment:
https://github.com/Samsung/ONE-vscode/pull/1315#discussion_r985182437

> What about separating 'one.viewer.metadata.show' command?
> This line is a bit tricky, as it does handle two cases: showing up in either `context menu in default explorer` or `context menu in ONE Explorer`.
> 
> How about adding another command for `default vscode explorer`, like 'one.viewer.metadata.showInDefaultExplorer'?
> Plus, if you agree, would you put the command title as 'ONE: Show Metadata', instead of 'Show Metadata' for the new command? Like, https://github.com/Samsung/ONE-vscode/pull/1315/files#r985182428

Same here! 😃

Reply:
That's a good way. I will apply the same as the Metadata Viewer.